client/ayon_core/hosts/traypublisher/plugins/create/create_csv_ingest.py

```diff
@@ -0,0 +1,722 @@
+from ayon_core.lib.transcoding import (
+    VIDEO_EXTENSIONS, IMAGE_EXTENSIONS
+)
+
+from ayon_core.hosts.traypublisher.api.plugin import (
+    TrayPublishCreator
+)
+
+
+class IngestCSV(TrayPublishCreator):
+    """CSV ingest creator class"""
+
+    icon = "fa.file"
+
+    label = "CSV Ingest"
+    product_type = "csv_ingest_file"
+    identifier = "io.ayon_core.creators.traypublisher.csv_ingest"
+
+    default_variants = ["Main"]
+
+    description = "Ingest products' data from CSV file"
+    detailed_description = """
+Ingest products' data from CSV file following column and representation
+configuration in project settings.
+"""
+
+    # Position in the list of creators.
+    order = 10
+
+    # settings for this creator
+    columns_config = {}
+    representations_config = {}
+
+    def create(self, subset_name, instance_data, pre_create_data):
+        """Create an product from each row found in the CSV.
+
+        Args:
+            subset_name (str): The subset name.
+            instance_data (dict): The instance data.
+            pre_create_data (dict):
+        """
+
+        csv_filepath_data = pre_create_data.get("csv_filepath_data", {})
+
+        folder = csv_filepath_data.get("directory", "")
+        if not os.path.exists(folder):
+            raise FileNotFoundError(
+                f"Directory '{folder}' does not exist."
+            )
+        filename = csv_filepath_data.get("filenames", [])
+        self._process_csv_file(subset_name, instance_data, folder, filename[0])
+
+    def _process_csv_file(
+            self, subset_name, instance_data, staging_dir, filename):
+        """Process CSV file.
+
+        Args:
+            subset_name (str): The subset name.
+            instance_data (dict): The instance data.
+            staging_dir (str): The staging directory.
+            filename (str): The filename.
+        """
+
+        # create new instance from the csv file via self function
+        self._pass_data_to_csv_instance(
+            instance_data,
+            staging_dir,
+            filename
+        )
+
+        csv_instance = CreatedInstance(
+            self.product_type, subset_name, instance_data, self
+        )
+        self._store_new_instance(csv_instance)
+
+        csv_instance["csvFileData"] = {
+            "filename": filename,
+            "staging_dir": staging_dir,
+        }
+
+        # from special function get all data from csv file and convert them
+        # to new instances
+        csv_data_for_instances = self._get_data_from_csv(
+            staging_dir, filename)
+
+        # create instances from csv data via self function
+        self._create_instances_from_csv_data(
+            csv_data_for_instances, staging_dir
+        )
+
+    def _create_instances_from_csv_data(
+        self,
+        csv_data_for_instances,
+        staging_dir
+    ):
+        """Create instances from csv data"""
+
+        for asset_name, _data in csv_data_for_instances.items():
```

Comment:
This should be `folder_path`, and I wonder why are `_data` with underscore?

Reply:
fixed